lib/workspace/src/usecase/render_image_for_export.dart

```diff
@@ -25,28 +25,51 @@ class RenderImageForExport {
       this._ref, this._graphicFactory, this._commandManager);
 
   Future<Image> call({bool keepTransparency = true}) async {
-    final recorder = _graphicFactory.createPictureRecorder();
-    final canvas = _graphicFactory.createCanvasWithRecorder(recorder);
+    final backgroundRecorder = _graphicFactory.createPictureRecorder();
+    final backgroundCanvas =
+        _graphicFactory.createCanvasWithRecorder(backgroundRecorder);
+
+    final canvasState = _ref.read(canvasStateProvider);
+    final exportSize = canvasState.size;
+
     if (!keepTransparency) {
       final paint = _graphicFactory.createPaint();
-      canvas.drawPaint(paint..color = const Color(0xFFFFFFFF));
+      backgroundCanvas.drawPaint(paint..color = const Color(0xFFFFFFFF));
     }
-    final canvasState = _ref.read(canvasStateProvider);
-    final exportSize = canvasState.size;
+
     final backgroundImage = canvasState.backgroundImage;
     final scaledRect = Rect.fromLTWH(0, 0, exportSize.width, exportSize.height);
     if (backgroundImage != null) {
       paintImage(
-        canvas: canvas,
+        canvas: backgroundCanvas,
         rect: scaledRect,
         image: backgroundImage,
         fit: BoxFit.fill,
         filterQuality: FilterQuality.none,
       );
     }
-    canvas.clipRect(scaledRect, doAntiAlias: false);
-    _commandManager.executeAllCommands(canvas);
-    final picture = recorder.endRecording();
+
+    final commandsRecorder = _graphicFactory.createPictureRecorder();
+    final commandsCanvas =
+        _graphicFactory.createCanvasWithRecorder(commandsRecorder);
+    commandsCanvas.clipRect(scaledRect, doAntiAlias: false);
+    _commandManager.executeAllCommands(commandsCanvas);
+
+    final combinedRecorder = _graphicFactory.createPictureRecorder();
+    final combinedCanvas =
+        _graphicFactory.createCanvasWithRecorder(combinedRecorder);
+
+    final bgPicture = backgroundRecorder.endRecording();
+    final bgImage = await bgPicture.toImage(
```

Comment:
I would write the whole name like you did with the others
backgroundPicture
backgroundImage
Also those 2 are synonyms, maybe specify what makes them different in the name

Reply:
Refactored the function, it should be more readable now